pyVmomi/VmomiSupport.py

```diff
@@ -460,6 +460,8 @@ def __getattr__(self, name):
    def __eq__(self, other):
       if other is None:
          return False
+      elif type(other) is str:
```

Comment:
I tested it against python3 and all worked as expected. Tried also using special characters, still everything is str object. 
@tianhao64 May be you can create a test that failes so I would know what to expect and add if anything is missing.